internal/auth0/connection/resource_clients_test.go

```diff
@@ -0,0 +1,145 @@
+package connection_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/auth0/terraform-provider-auth0/internal/acctest"
+)
+
+func TestAccConnectionClientsPreventErasingEnabledClientsOnCreate(t *testing.T) {
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: `
+resource "auth0_connection" "my_conn" {
+	name = "Acceptance-Test-Connection-PreventErasing"
+	strategy = "auth0"
+}
+
+resource "auth0_client" "my_client" {
+	depends_on = [ auth0_connection.my_conn ]
+	name = "Acceptance-Test-Client-1-PreventErasing"
+}
+
+# Pre-existing enabled client
+resource "auth0_connection_client" "one_to_one" {
+	depends_on = [ auth0_client.my_client ]
+	connection_id = auth0_connection.my_conn.id
+	client_id = auth0_client.my_client.id
+}
+
+resource "auth0_connection_clients" "one_to_many" {
+	depends_on = [ auth0_connection_client.one_to_one ]
+	connection_id = auth0_connection.my_conn.id
+	enabled_clients = []
+}
+`,
+				ExpectError: regexp.MustCompile("Connection with non empty enabled clients"),
+			},
+		},
+	})
+}
+
+const givenASingleConnection = `
+resource "auth0_connection" "my_conn" {
+	name = "Acceptance-Test-Connection-{{.testName}}"
+	strategy = "auth0"
+}
+`
+
+const testAccConnectionClientsWithMinimalConfig = givenASingleConnection + `
+resource "auth0_connection_clients" "my_conn_client_assoc" {
+	depends_on = [ auth0_connection.my_conn ]
+
+	connection_id = auth0_connection.my_conn.id
+	enabled_clients = []
+}
+`
+
+const testAccConnectionClientsWithOneEnabledClient = givenASingleConnection + `
+resource "auth0_client" "my_client-1" {
+	depends_on = [ auth0_connection.my_conn ]
+
+	name = "Acceptance-Test-Client-1-{{.testName}}"
+}
+
+resource "auth0_connection_clients" "my_conn_client_assoc" {
+	depends_on = [ auth0_client.my_client-1 ]
+
+	connection_id = auth0_connection.my_conn.id
+	enabled_clients = [auth0_client.my_client-1.id]
+}
+`
+
+const testAccConnectionClientsWithTwoEnabledClients = givenASingleConnection + `
+resource "auth0_client" "my_client-1" {
+	depends_on = [ auth0_connection.my_conn ]
+
+	name = "Acceptance-Test-Client-1-{{.testName}}"
+}
+
+resource "auth0_client" "my_client-2" {
+	depends_on = [ auth0_client.my_client-1 ]
+
+	name = "Acceptance-Test-Client-1-{{.testName}}"
+}
+
+resource "auth0_connection_clients" "my_conn_client_assoc" {
+	depends_on = [ auth0_client.my_client-2 ]
+
+	connection_id = auth0_connection.my_conn.id
+	enabled_clients = [auth0_client.my_client-1.id, auth0_client.my_client-2.id]
+}
+`
+
+const testAccConnectionClientsWithNoEnabledClients = givenASingleConnection + `
+resource "auth0_connection_clients" "my_conn_client_assoc" {
+	depends_on = [ auth0_connection.my_conn ]
+
+	connection_id = auth0_connection.my_conn.id
+	enabled_clients = []
+}
+`
```

Comment:
Idea: test the deletion of this resource by using a connection data source to check how many enabled clients there are.

Reply:
Followed up in https://github.com/auth0/terraform-provider-auth0/pull/570.